hazelcast/src/main/java/com/hazelcast/internal/config/override/ExternalConfigurationOverride.java

```diff
@@ -0,0 +1,89 @@
+package com.hazelcast.internal.config.override;
+
+import com.hazelcast.client.config.ClientConfig;
+import com.hazelcast.client.config.impl.ClientDomConfigProcessor;
+import com.hazelcast.config.Config;
+import com.hazelcast.internal.config.MemberDomConfigProcessor;
+import com.hazelcast.logging.ILogger;
+import com.hazelcast.logging.Logger;
+
+import java.util.Arrays;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Set;
+import java.util.function.BiConsumer;
+
+import static java.lang.String.format;
+
+public class ExternalConfigurationOverride<T> {
+
+    private static final ILogger LOGGER = Logger.getLogger(ExternalConfigurationOverride.class);
+
+    private final Set<ConfigProvider> configProviders;
+
+    private final BiConsumer<ConfigProvider, T> configProcessor;
+
+    ExternalConfigurationOverride(BiConsumer<ConfigProvider, T> configProcessor, ConfigProvider... providers) {
+        this.configProcessor = configProcessor;
+        this.configProviders = new HashSet<>(Arrays.asList(providers));
+    }
+
+    public static ExternalConfigurationOverride<Config> member() {
+        return new ExternalConfigurationOverride<>(
+          memberConfigProcessor(),
+          new EnvConfigProvider(EnvVariablesConfigParser.member()),
+          new SystemPropertiesConfigProvider(SystemPropertiesConfigParser.member()));
+    }
+
+    public static ExternalConfigurationOverride<ClientConfig> client() {
+        return new ExternalConfigurationOverride<>(
+          clientConfigProcessor(),
+          new EnvConfigProvider(EnvVariablesConfigParser.client()),
+          new SystemPropertiesConfigProvider(SystemPropertiesConfigParser.client()));
+    }
+
+    public T overwrite(T config) {
+        ConfigValidator.validate(configProviders);
+
+        for (ConfigProvider configProvider : configProviders) {
+            Map<String, String> properties = configProvider.properties();
+
+            if (properties.size() > 0) {
```

Comment:
The problem is that we don't have any way to validate if an entry is a part of the configuration or not. It's like entering a non-existing entry into a YAML file. 

Our configuration is essentially a huge tree used for lookups by individual parsers, validated only by the XML schema.

Naturally, we can exclude existing config entries (like the Dockerfile ones), but I'm afraid this would need to be done manually at the filtering stage.